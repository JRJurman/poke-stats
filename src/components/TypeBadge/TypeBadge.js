import "./TypeBadge.scss"
import { registerHtml } from "tram-one"

const html = registerHtml()

export default (props) => {
  const displayName = props.short ? props.type.slice(0,3) : props.type;
  if (props.vertical) {
    return html`
    <div class="TypeBadge vertical-container">
      <div class=${`TypeBadgeTop type-${props.type}`}> ${displayName} </div>
      <div class=${`TypeBadgeBottom level-${props.effectiveness}`}></div>
    </div>
    `
  }
  if (props.effectiveness) {
    return html`
    <div class="TypeBadge horizontal-container">
      <div class=${`TypeBadgeLeft type-${props.type}`}> ${displayName} </div>
      <div class=${`TypeBadgeRight level-${props.effectiveness}`}></div>
    </div>
    `
  }
  return html`
    <span class=${`TypeBadge type-${props.type}`}>${displayName}</span>
  `
}
