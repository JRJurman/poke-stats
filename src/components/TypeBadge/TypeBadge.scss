
.TypeBadge, .horizontal-container, .vertical-container {
  padding:5px;
  margin:5px;
  border-radius: 8px;
  border: 2px solid rgba(0,0,0,0.25);
  font-family: monospace;
  color: white;
  text-transform: uppercase;
  font-weight: bold;
  text-shadow: 2px 2px 5px black;
}

.horizontal-container {
  float: left;
  padding: 0px;
  border: none;
  display: inline-flex;
  
  .level-0::after {
    content: '0';
  }
  .level-0\.25::after {
    content: '¼'
  }
  .level-0\.5::after {
    content: '½'
  }
  .level-2::after {
    content: '2'
  }
  .level-4::after {
    content: '4'
  }
}

.TypeBadgeLeft {
  border-right: 0px;
  margin-right: 0px;
  border-radius: 8px 0px 0px 8px;
  padding: 5px;
  border: 2px solid rgba(0,0,0,0.25);
}

.TypeBadgeRight {
  border-left: 0px;
  margin-left: 0px;
  border-radius: 0px 8px 8px 0px;
  padding: 5px;
  border: 2px solid rgba(0,0,0,0.25);
}

.vertical-container {
  text-align: center;
  padding: 0px;
  border: none;
  float: left;

  .level-0::after {
    content: 'NO EFFECT';
  }
  .level-0\.25::after {
    content: 'VERY NOT EFFECTIVE'
  }
  .level-0\.5::after {
    content: 'NOT EFFECTIVE'
  }
  .level-2::after {
    content: 'SUPER EFFECTIVE'
  }
  .level-4::after {
    content: 'VERY SUPER EFFECTIVE'
  }
}

.TypeBadgeTop {
  border-right: 0px;
  margin-right: 0px;
  border-radius: 8px 8px 0px 0px;
  border: 2px solid rgba(0,0,0,0.25);
  padding: 3px;
  padding-bottom: 0px;
}

.TypeBadgeBottom {
  border-left: 0px;
  margin-left: 0px;
  border-radius: 0px 0px 8px 8px;
  border: 2px solid rgba(0,0,0,0.25);
  padding: 3px;
  padding-top: 0px;
}

.level-0 {
  background: #2e3436;
}

.level-0\.25 {
  background: #7c0000;
}

.level-0\.5 {
  background: #a40000;
}

.level-2 {
  background: #4e9a06;
}

.level-4 {
  background: #73d216;
}

.type-normal{background-color:#aa9}
.type-fire{background-color:#f42}
.type-water{background-color:#39f}
.type-electric{background-color:#fc3}
.type-grass{background-color:#7c5}
.type-ice{background-color:#6cf}
.type-fighting{background-color:#b54}
.type-poison{background-color:#a59}
.type-ground{background-color:#db5}
.type-flying{background-color:#89f}
.type-psychic{background-color:#f59}
.type-bug{background-color:#ab2}
.type-rock{background-color:#ba6}
.type-ghost{background-color:#66b}
.type-dragon{background-color:#76e}
.type-dark{background-color:#754}
.type-steel{background-color:#aab}
.type-fairy{background-color:#e9e}
