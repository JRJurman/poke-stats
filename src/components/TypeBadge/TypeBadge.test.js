import TypeBadge from './TypeBadge';

describe('TypeBadge', () => {
  it('should match snapshot with type', () => {
    const wrapper = TypeBadge({ type: 'FIRE' });
    expect(wrapper.outerHTML).toMatchSnapshot();
  });

  it('should match snapshot with type in shorthand', () => {
    const wrapper = TypeBadge({ type: 'FIRE', short: true });
    expect(wrapper.outerHTML).toMatchSnapshot();
  });

  it('should match snapshot with type and effectiveness', () => {
    const wrapper = TypeBadge({ type: 'FIRE', effectiveness: 2 });
    expect(wrapper.outerHTML).toMatchSnapshot();
  });

  it('should match snapshot with type and effectiveness in vertical', () => {
    const wrapper = TypeBadge({ type: 'FIRE', effectiveness: 2, vertical: true });
    expect(wrapper.outerHTML).toMatchSnapshot();
  });
});
