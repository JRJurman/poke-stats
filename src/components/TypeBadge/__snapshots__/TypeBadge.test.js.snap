// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeBadge should match snapshot with type 1`] = `"<span class=\\"TypeBadge type-FIRE\\">FIRE</span>"`;

exports[`TypeBadge should match snapshot with type and effectiveness 1`] = `"<div class=\\"horizontal-container\\"><div class=\\"TypeBadgeLeft type-FIRE\\">FIRE</div><div class=\\"TypeBadgeRight level-2\\"></div></div>"`;

exports[`TypeBadge should match snapshot with type and effectiveness in vertical 1`] = `"<div class=\\"vertical-container\\"><div class=\\"TypeBadgeTop type-FIRE\\">FIRE</div><div class=\\"TypeBadgeBottom level-2\\"></div></div>"`;

exports[`TypeBadge should match snapshot with type in shorthand 1`] = `"<span class=\\"TypeBadge type-FIRE\\">FIR</span>"`;
